platform/security/platform-security-core-api/src/main/java/ddf/security/samlp/SimpleSign.java

```diff
@@ -317,6 +303,53 @@ private void validateSignatureAndSamlKey(Signature signature, SAMLKeyInfo samlKe
     }
   }
 
+  private void signSamlObject(
+      SignableSAMLObject samlObject, String sigAlgo, String canonAlgo, String digestAlgo)
+      throws SignatureException {
+    X509Certificate[] certificates = getSignatureCertificates();
+    PrivateKey privateKey = getSignaturePrivateKey();
+
+    // Create the signature
+    Signature signature = OpenSAMLUtil.buildSignature();
+    if (signature == null) {
+      throw new SignatureException("Unable to build signature.");
+    }
+
+    signature.setCanonicalizationAlgorithm(canonAlgo);
+    signature.setSignatureAlgorithm(sigAlgo);
+
+    BasicX509Credential signingCredential = new BasicX509Credential(certificates[0]);
+    signingCredential.setPrivateKey(privateKey);
+
+    signature.setSigningCredential(signingCredential);
+
+    X509KeyInfoGeneratorFactory x509KeyInfoGeneratorFactory = new X509KeyInfoGeneratorFactory();
+    x509KeyInfoGeneratorFactory.setEmitEntityCertificate(true);
+
+    try {
+      KeyInfo keyInfo = x509KeyInfoGeneratorFactory.newInstance().generate(signingCredential);
+      signature.setKeyInfo(keyInfo);
+    } catch (org.opensaml.security.SecurityException e) {
+      throw new SignatureException("Error generating KeyInfo from signing credential", e);
+    }
+
+    if (samlObject instanceof Response) {
+      List<Assertion> assertions = ((Response) samlObject).getAssertions();
+      for (Assertion assertion : assertions) {
+        assertion.getSignature().setSigningCredential(signingCredential);
```

Comment:
❓  Since the result of `getSignature()` is `@Nullable` should we account for that here?